app/tournament/match.js

```diff
@@ -0,0 +1,20 @@
+export class Match {
+    constructor() {
+        this.round = 1;
+        this.players = [];
+        this.parents = [];
+    }
+
+    addPlayer(player) {
+        if (this.players.length >= 2) {
+            throw new MatchException("Match has already 2 players");
+        }
+        this.players.push(player);
+    }
+}
+
+export class MatchException extends Match {
```

Comment:
This is very wrong, not just that I don´t feel we need an error class, it´s that it´s extending from Match.